lua/feline/generator.lua

```diff
@@ -230,25 +229,25 @@ local function parse_statusline_section_old(section, type)
 end
 
 -- Parse components of a section of the statusline
-local function parse_statusline_section(section)
+local function parse_statusline_section(section, winnr)
     local components = {}
 
     for _, component in ipairs(section) do
-        components[#components+1] = parse_component(component)
+        components[#components+1] = parse_component(component, winnr)
     end
 
     return table.concat(components)
 end
 
 -- Generate statusline by parsing all components and return a string
-function M.generate_statusline(is_active)
+function M.generate_statusline(winnr)
     if not M.components then
-        return ""
+        return ''
     end
 
     local statusline_type
 
-    if is_active and not is_forced_inactive() then
+    if winnr == vim.api.nvim_get_current_win() and not is_forced_inactive() then
```

Comment:
This will always evaluate to false since winnr is nil

Reply:
Added default value for winnr to work around that for now. Once this is merged I'll make proper use and actually pass the window number to the statusline generator